zerver/lib/test_classes.py

```diff
@@ -327,12 +327,14 @@ def submit_reg_form_for_user(
     def get_confirmation_url_from_outbox(self, email_address: Text, *,
                                          url_pattern: Text=None) -> Text:
         from django.core.mail import outbox
-        if url_pattern is None:
+        if url_pattern is not None:
+            url_pattern_to_use = url_pattern
+        else:
             # This is a bit of a crude heuristic, but good enough for most tests.
-            url_pattern = settings.EXTERNAL_HOST + "(\S+)>"
+            url_pattern_to_use = settings.EXTERNAL_HOST + "(\S+)>"
         for message in reversed(outbox):
             if email_address in message.to:
-                return re.search(url_pattern, message.body).groups()[0]
+                return re.search(url_pattern_to_use, message.body).groups()[0]
```

Comment:
This hunk feels a bit messy; is there a better way to do this?

Reply:
Do you just mean this commit? Messy how?

The issue in this commit is that `url_pattern`  is `Optional[Text]`, whereas `re.search` accepts an `AnyStr`. I didn't want to rename the function parameter, so introduced another name internally; a different internal one would be fine if it read better. I also reordered the conditional - is that the particular part that seems messy?